File: csharp/剑指32从上到下打印二叉树.cs
public partial class Solution
{
    public class TreeNode
    {
        public int val;
        public TreeNode left;
        public TreeNode right;
        public TreeNode(int x) { val = x; }
    }

    public int[] LevelOrder(TreeNode root)
    {
        List<int> res = new ();
        Queue<TreeNode> nodes = new();
        nodes.Enqueue(root);
        
        while(nodes.Count>0)
        {
            TreeNode crt = nodes.Dequeue();
            if(crt!=null)
            {
                res.Add(crt.val);
                nodes.Enqueue(crt.left);
                nodes.Enqueue(crt.right);
            }
        }
            
        return res.ToArray();
    }
}

File: python/main/剑指32从上到下打印二叉树.py
from binarytree import TreeNode
from collections import deque

def levelOrder(root: TreeNode) -> list:
    if(root==None):
        return []
    nodes=deque([root])
    flag=True
    res=[]
    while(len(nodes)):
        size=len(nodes)
        temp=[]
        for _ in range(size):
            crt=nodes.popleft()
            if(crt!=None):
                nodes.append(crt.left)
                nodes.append(crt.right)
                temp.append(crt.val)
        if(len(temp)):
            res.append(temp if flag else list(reversed(temp)))
        flag=not flag
        
    return res